packages/@react-spectrum/test-utils/src/renderOverride.tsx

```diff
@@ -28,9 +30,18 @@ if (!renderHook) {
   actHook = rhtl.act;
 }
 
-function customRender(ui, options) {
-  return render(ui, {wrapper: StrictModeWrapper, ...options});
+interface RenderOptions extends BaseRenderOptions {
+  providerProps: Omit<ProviderProps, 'children'>
 }
 
+// TODO: perhaps export this as a non-overide of render?
+function providerWrapperRender(ui: ReactElement, options?: RenderOptions) {
+  let rendered = render(ui, {wrapper: (props) => <ProviderWrapper {...props} {...options?.providerProps} />, ...options});
+  return {
+    ...rendered,
+    rerender: (ui, options) => providerWrapperRender(ui, {container: rendered.container, ...options})
+  };
+}
```

Comment:
redefining `rerender` so that the user can provide new `providerProps` seems to cause previously queried elements to be come stale (aka `let button = getByRole('button');` will point to the pre-rerender version of button). Going to move this work to a [different branch ](https://github.com/adobe/react-spectrum/tree/provider_render_wrapper) to work on separately, prioritizing other utils first